### PR TITLE
KEP for allowing Addon Resizer 1.8 to scale based on container count

### PR DESCRIPTION
**Pull request by jbartosik**
/kind documentation
/kind feature

#5546

### CODE REVIEW

**Comment by jbartosik** [comment]
/hold work-in-progress

I want to make some changes before starting review

File: addon-resizer/enhancements/5546-scaling-based-on-container-count/README.md [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+
+### Goals
+
+- Allow scaling workloads based on count of containers in a cluster.
+- Allow this for Addon Resizer 1.8 ([used by metrics  server]).
+
+### Non-Goals
+
+- Using both node and container count to scale workloads.
+- Bringing this change to the `master` branch of Addon Resizer.
```
[the remark this answers is not in the document]
Comment (jbartosik) [reply addon-resizer/enhancements/5546-scaling-based-on-container-count/README.md:28]: Yes. Addon resizer on the `master` branch is very different, we'd need to basically implement this twice. And we're not really doing releases from `master` anyways.

File: addon-resizer/enhancements/5546-scaling-based-on-container-count/README.md [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+
+- [Flag descriptions] that directly reference node counts (`--extra-cpu`, `--extra-memory`, `--extra-storage`, and
+  `--minClusterSize`) will need to be updated to instead refer to cluster size.
+- [README] will need to be updated to reference cluster size instead of node count and explain that cluster size refers
+  to either node count or container count, depending on the value of the `--scaling-mode` flag.
+- Many variable names in code which now refer to node count will refer to cluster size and should be renamed accordingly.
+
+In addition to implementing the feature we should also clean up the code and documentation.  
+
+### Risks and Mitigations
```
[the remark this answers is not in the document]
Comment (jbartosik) [reply addon-resizer/enhancements/5546-scaling-based-on-container-count/README.md:51]: I think setting [`estimator`](https://github.com/kubernetes/autoscaler/blob/da500188188d275a382be578ad3d0a758c3a170f/addon-resizer/nanny/main/pod_nanny.go#L62) to `exponential` would help, I'll check carefully tommorow

**Review by mwielgus — APPROVED** [review]
/lgtm
/approve

**Pull request merged**